lms/djangoapps/course_home_api/outline/serializers.py

```diff
@@ -57,10 +64,27 @@ def get_blocks(self, block):  # pylint: disable=missing-function-docstring
                 'hide_from_toc': block.get('hide_from_toc'),
             },
         }
+        if 'special_exam_info' in self.context.get('extra_fields', []) and block.get('special_exam_info'):
+            serialized[block_key]['special_exam_info'] = block.get('special_exam_info').get('short_description')
+
         for child in children:
             serialized.update(self.get_blocks(child))
         return serialized
 
+    @staticmethod
+    def get_vertical_icon_class(block):
+        """
+        Get the icon class for a vertical block based on its children.
+        """
+        children = block.get('children', [])
+        child_classes = {child.get('type') for child in children}
+        new_class = 'other'
+        icon_call_priority = ['video', 'problem']
+        for higher_class in icon_call_priority:
+            if higher_class in child_classes:
+                new_class = higher_class
+        return new_class
```

Comment:
Nit: When I first read `icon_call_priority`, I assumed that the things that came first had higher priority, rather than the things that came last. I think this might read more clearly if you use returns instead of a `new_class` var, so something like:

```
    @staticmethod
    def get_vertical_icon_class(block):
        """
        Get the icon class for a vertical block based on its children.
        """
        children = block.get('children', [])
        child_classes = {child.get('type') for child in children}
        icon_call_priority = ['problem', 'video']
        for item_type in icon_call_priority:
            if item_type in child_classes:
                 return item_type
        return 'other'  # default
```

You could also make it more explicit/obvious like this, since there are only a few classes that matter at the moment:

```
    @staticmethod
    def get_vertical_icon_class(block):
        """
        Get the icon class for a vertical block based on its children.
        """
        children = block.get('children', [])
        child_classes = {child.get('type') for child in children}
        if 'problem' in child_classes:
            return 'problem'
        if 'video' in child_classes:
            return 'video'
        return 'other'
```

Either way, please describe the intended ordering in the docstring.

Reply:
Fixed